BST.rb

```diff
@@ -13,19 +14,21 @@ def initialize(value = nil, left = nil, right = nil)
 
 
 class BinarySearchTree
+
+    attr_accessor :root
```

Comment:
If you have added this, change all the `@root` to `root` and any `root` inside a function can be `_root` or `curr_root`.

Reply:
https://www.rubyguides.com/2018/11/attr_accessor/

Reply:
after adding attr_accessor, on changing @root -> root, breaks the working of code. need to discuss that with you.